File: C++/Level1/food fighter.cpp
/*
	https://school.programmers.co.kr/learn/courses/30/lessons/134240
*/

#include <string>
#include <vector>

using namespace std;

string solution(vector<int> food) {
    string answer = "";
    string right = "";

    for (int i = 1; i < food.size(); ++i)
    {
        answer += string(food[i] / 2, to_string(i)[0]);
        right = string(food[i] / 2, to_string(i)[0]) + right;
    }

    return answer + "0" + right;
}

File: Java/Level1/Food Fighter.java
/*
 * https://school.programmers.co.kr/learn/courses/30/lessons/134240?language=java
 */

class Solution {
    public String solution(int[] food) {
        String left = "", right = "";
        
        for(int i = 1; i < food.length; ++i)
        {
            int foodCount = food[i] / 2;

            String foodString = new String(Integer.toString(i)).repeat(foodCount);
            left += foodString;
            right = foodString + right;
        }

        return left + '0' + right;
    }
}